### PR TITLE
GH-115869: Make `jit_stencils.h` reproducible

### PR DESCRIPTION
**Pull request by brandtbucher**
At the core dev sprint, we discussed the possibility of hosting pre-built `jit_stencils.h` files for common platforms and build configurations.

While there's a lot of work involved in making that happen (I'll post a rough plan shortly), this is the first logical step in that direction. This PR replaces absolute file paths in the generated code with relative ones, and processes stencils in sorted order.

* Issue: gh-115869

### CODE REVIEW

File: Tools/jit/_stencils.py [hunk 1]
```diff
@@ -202,7 +202,8 @@ def pad(self, alignment: int) -> None:
         """Pad the stencil to the given alignment."""
         offset = len(self.body)
         padding = -offset % alignment
-        self.disassembly.append(f"{offset:x}: {' '.join(['00'] * padding)}")
+        if padding:
```
Comment (brandtbucher) [inline Tools/jit/_stencils.py:205]: This is an unrelated cleanup.

**Review by mdboom — APPROVED** [review]

**Review by savannahostrowski — APPROVED** [review]
LGTM!
